jax/ops/scatter.py

```diff
@@ -327,7 +329,8 @@ def segment_sum(data, segment_ids, num_segments=None,
     segment sums.
   """
   if num_segments is None:
-    num_segments = jnp.max(jnp.mod(segment_ids, data.shape[0])) + 1
+    num_segments = jnp.max(
```

Comment:
I think you can just write this as `max(jnp.max(segment_ids) + 1, jnp.max(-segment_ids))`.

Reply:
I did a quick check and I seem to get this error if using the built in max within the jitted function.

```
import jax
import jax.numpy as jnp

@jax.jit
def fn(a):
  return max(jnp.max(a) + 1, jnp.max(-a))


input = np.array([1, 2, 3])
fn(input)
```
> ConcretizationTypeError: Abstract tracer value encountered where concrete value is expected (in `bool`).
> Use transformation parameters such as `static_argnums` for `jit` to avoid tracing input values.
> See `https://jax.readthedocs.io/en/latest/faq.html#abstract-tracer-value-encountered-where-concrete-value-is-expected-error`.
> Encountered value: Traced<ShapedArray(bool[]):JaxprTrace(level=-1/1)>

Reply:
Yes, but in this specific circumstance, you are about to call `int(...)` on the output of this function. So you'd get that exact error message anyway under a `jit`, and this is what the comment means when it says you must provide this value explicitly under a `jit`.

Reply:
Done.